cpp/src/cylon/arrow/arrow_kernels.cpp

```diff
@@ -42,37 +40,50 @@ class ArrowArrayNumericSplitKernel : public ArrowArraySplitKernel {
       return Status(Code::ExecutionError, "values rows != target_partitions length");
     }
 
-    std::vector<std::unique_ptr<ARROW_BUILDER_T>> builders;
-    builders.reserve(num_partitions);
+    std::vector<std::shared_ptr<arrow::Buffer>> build_buffers;
+    std::vector<T *> data_buffers;
+    std::vector<int> buffer_indexes;
```

Comment:
I feel like ` std::vector<int> buffer_indexes` is redundant.

Reply:
I didn't get it

Reply:
is `buffer_indexes` used somewhere?